src/overlay/FlowControl.cpp

```diff
@@ -171,22 +126,23 @@ FlowControl::maybeSendNextBatch()
             // Can't send _current_ message
             if (!hasOutboundCapacity(msg))
             {
-                CLOG_DEBUG(Overlay, "{}: No outbound capacity for peer {}",
-                           mApp.getConfig().toShortString(
-                               mApp.getConfig().NODE_SEED.getPublicKey()),
-                           mApp.getConfig().toShortString(mNodeID));
+                CLOG_DEBUG(
+                    Overlay, "{}: No outbound capacity for peer {}",
+                    mAppConnector.getConfig().toShortString(
+                        mAppConnector.getConfig().NODE_SEED.getPublicKey()),
+                    mAppConnector.getConfig().toShortString(mNodeID));
                 // Start a timeout for SEND_MORE
                 mNoOutboundCapacity =
                     std::make_optional<VirtualClock::time_point>(
-                        mApp.getClock().now());
+                        mAppConnector.now());
                 break;
             }
 
-            mSendCallback(msg);
+            mSendCallback(std::make_shared<StellarMessage>(msg));
```

Comment:
Good idea to avoid any additional copying. I updated this path to take `front.mMessage` instead, and also realized that we're doing a lot of copying in our transaction/SCP flooding paths (this was the case for a while). I updated those places as well. Thanks for the suggestion!